src/librustc_lexer/Cargo.toml

```diff
@@ -7,3 +7,7 @@ edition = "2018"
 # Note that this crate purposefully does not depend on other rustc crates
 [dependencies]
 unicode-xid = { version = "0.1.0", optional = true }
+
+[lib]
+doctest = false
+name = "rustc_lexer"
```

Comment:
Could you add a comment explaining that we need this "no-op" lib section for purely for ap?

Reply:
Added a comment in https://github.com/rust-lang/rust/pull/63036/commits/98f29f5e38bc385f661fc81d67f980bc0e91d6e4.